### PR TITLE
bump transformers and update attention class map name

### PR DESCRIPTION
**Pull request by winglian**
bump transformers to catches fixes 28133 and 28115 for mistral/mixtral. 28133 changed a variable name that we use in a patch too.

### CODE REVIEW

**Review by hamelsmu — APPROVED** [review]

File: src/axolotl/monkeypatch/mixtral/__init__.py
Comment (ehartford) [inline src/axolotl/monkeypatch/mixtral/__init__.py:1]: transformers.models.mixtral.modeling_mixtral.MixtralModel._use_flash_attention_2 = (
        True
    )

Comment (winglian) [reply src/axolotl/monkeypatch/mixtral/__init__.py:1]: ???

Comment (ehartford) [reply src/axolotl/monkeypatch/mixtral/__init__.py:1]: You gotta add that, to make it work.

Comment (ehartford) [reply src/axolotl/monkeypatch/mixtral/__init__.py:1]: https://discord.com/channels/1104757954588196865/1117071926926512248/1190577865792688271

Comment (ehartford) [reply src/axolotl/monkeypatch/mixtral/__init__.py:1]: else you get this error

Comment (winglian) [reply src/axolotl/monkeypatch/mixtral/__init__.py:1]: we already handle that correctly here https://github.com/OpenAccess-AI-Collective/axolotl/blob/main/src/axolotl/utils/models.py#L326-L346

**Comment by tcapelle** [comment]
Can we merge this and re build docker please =)

**Comment by winglian** [comment]
added some additional tests for mixtral and mistral to verify that things get patched properly.

**Review by hamelsmu — APPROVED** [review]
LGTM thanks for adding the tests

**Pull request merged**